core/src/main/java/org/elasticsearch/cluster/ClusterStateTaskExecutor.java

```diff
@@ -120,7 +120,7 @@ private TaskResult(Throwable failure) {
         }
 
         public boolean isSuccess() {
-            return failure != null;
```

Comment:
This was a bug, but fortunately the method was unused until now so not impactful.